### ISSUE TITLE
Authentication fails: Bad Request (Invalid host)

### ISSUE THREAD

**Issue by eusogames** [danqzq/unity-twitch-chat-interactions#9]
After loading the authentication scene and attempting to authenticate, the twitch authentication window opens, and after authorizing it, It fails, saying Bad Request (Invalid host). Further attempts take me directly to the Bad Request (Invalid host) page.

**Comment by eusogames** [comment]
TwitchAuthenticator.cs Line 67 http://localhost/ removing the extra slash from the end of http://localhost seems to have fixed the issue as is shown in [](https://dev.twitch.tv/docs/irc/authenticate-bot/). The issues appears to have been redirect_uri does not match registered URI.

**Comment by danqzq** [comment]
Removed the extra slash from the localhost URL in TwitchAuthenticator.cs

**Comment by eusogames** [comment]
Sorry meant to update this sooner, sent this before i could test this and removing the slash does create a no path specified error. I was experimenting and believed the issue to be a mismatched uri but after more testing this doesn't appear to be the issue. It seems to be with retrieving the token from the request.

**Comment by danqzq** [comment]
May I know which browser you are using?

**Comment by eusogames** [comment]
Currently using Firefox. Tried disabling extensions in case that was causing an issue but haven't tried with a different browser yet.

**Comment by eusogames** [comment]
After switching browser to chrome, authentication completes correctly.

**Comment by danqzq** [comment]
Yeah, users have reported issues with the Firefox browser not proceeding the authentication.

**Comment by Thesie** [comment]
Unfortunately I have the exact same bug still (also on Firefox). It gave me the correct prompt the first time, after that I am taken instantly to the "Bad Request (Invalid host)". I disabled the extensions, this did not help =(

**Comment by danqzq** [comment]
Nudging in, @vlandemart I believe you've made the necessary changes in your fork. Ready to make a PR?

**Comment by vlandemart** [comment]
Sorry, I've had some compiling erros and was too lazy to boot up project to fix them. Now should be all good.
